drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -0,0 +1,893 @@
+	 */
+	} else {
+		if (channel < 11 || channel > 26) {
+			LOG_ERR("Unsupported channel %u", channel);
+			return -EINVAL;
+		}
+
+		u8_t reg = rf2xx_iface_reg_read(dev, PHY_CC_CCA_REG) & ~0x1f;
+		rf2xx_iface_reg_write(dev, PHY_CC_CCA_REG, reg | channel);
+	}
+
+	return 0;
+}
+
+static int rf2xx_set_txpower(struct device *dev, s16_t dbm)
+{
+	(void) dbm;
+	/* TODO: Add look-up table
+	 * Now will max power
+	 */
+	u8_t reg = rf2xx_iface_reg_read(dev, PHY_TX_PWR_REG) & ~0x0f;
+	rf2xx_iface_reg_write(dev, PHY_TX_PWR_REG, reg);
+
+	return 0;
+}
+
+static int rf2xx_set_ieee_addr(struct device *dev, bool set,
+	const u8_t *ieee_addr)
+{
+	LOG_DBG("IEEE address %02x:%02x:%02x:%02x:%02x:%02x:%02x:%02x",
+		ieee_addr[7], ieee_addr[6], ieee_addr[5], ieee_addr[4],
+		ieee_addr[3], ieee_addr[2], ieee_addr[1], ieee_addr[0]);
+
+	if (set) {
+		const u8_t *ptr_to_reg = ieee_addr;
+		for (u8_t i = 0; i < 8; i++) {
+			rf2xx_iface_reg_write(dev, (IEEE_ADDR_0_REG + i),
+				*ptr_to_reg);
+			ptr_to_reg++;
+		}
+	} else {
+		for (u8_t i = 0; i < 8; i++) {
+			rf2xx_iface_reg_write(dev, (IEEE_ADDR_0_REG + i), 0);
+		}
+	}
+	return 0;
+}
+static int rf2xx_set_short_addr(struct device *dev, bool set,
+	u16_t short_addr)
+{
+	u8_t short_addr_le[2] = { 0xFF, 0xFF };
+
+	if (set) {
+		sys_put_le16(short_addr, short_addr_le);
+	}
+
+	rf2xx_iface_reg_write(dev, SHORT_ADDR_0_REG, short_addr_le[0]);
+	rf2xx_iface_reg_write(dev, SHORT_ADDR_1_REG, short_addr_le[1]);
+	rf2xx_iface_reg_write(dev, CSMA_SEED_0_REG,
+			      short_addr_le[0] + short_addr_le[1]);
+
+	if (set) {
+		LOG_DBG("0x%x", short_addr);
+	} else {
+		LOG_DBG("0xFFFF");
+	}
+
+	return 0;
+}
+static int rf2xx_set_pan_id(struct device *dev, bool set, u16_t pan_id)
+{
+	u8_t pan_id_le[2] = { 0xFF, 0xFF };
+
+	if (set) {
+		sys_put_le16(pan_id, pan_id_le);
+	}
+
+	rf2xx_iface_reg_write(dev, PAN_ID_0_REG, pan_id_le[0]);
+	rf2xx_iface_reg_write(dev, PAN_ID_1_REG, pan_id_le[1]);
+
+	if (set) {
+		LOG_DBG("0x%x", pan_id);
+	} else {
+		LOG_DBG("0xFFFF");
+	}
+
+	return 0;
+}
+
+static int rf2xx_filter(struct device *dev,
+	bool set, enum ieee802154_filter_type type,
+	const struct ieee802154_filter *filter)
+{
+	LOG_DBG("Applying filter %u", type);
+
+	if (type == IEEE802154_FILTER_TYPE_IEEE_ADDR) {
+		return rf2xx_set_ieee_addr(dev, set, filter->ieee_addr);
+	} else if (type == IEEE802154_FILTER_TYPE_SHORT_ADDR) {
+		return rf2xx_set_short_addr(dev, set, filter->short_addr);
+	} else if (type == IEEE802154_FILTER_TYPE_PAN_ID) {
+		return rf2xx_set_pan_id(dev, set, filter->pan_id);
+	}
+
+	return -ENOTSUP;
+}
+
+static int rf2xx_tx(struct device *dev,
+	struct net_pkt *pkt,
+	struct net_buf *frag)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	bool abort = true;
+	int response;
+
+	k_mutex_lock(&ctx->phy_mutex, K_FOREVER);
+	/* Reset semaphore in case ACK was received after timeout */
+	k_sem_reset(&ctx->trx_tx_sync);
+	if (ctx->trx_state == RF2XX_TRX_PHY_STATE_IDLE) {
```

Comment:
have an empty line before if/for/while unless it's a variable allocation and it is actually being used in the conditional statement

Reply:
under review

Reply:
Fixed